docs/http-api-integration.md

```diff
@@ -0,0 +1,112 @@
+
+## Prerequisite Understanding and Terms
+As outlined by [Roy Fielding](https://www.ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm) (citation), the **Six Architectural Constraints for RESTful API** are as follows:
+1. Client-Server
+2. Stateless
+3. Cache
+4. Uniform Interface
+5. Layered System
+6. Code-On-Demand
+
+
+## Limitations of the Constraints
+A supplemental specification is necessary beyond the _six architectural constraints_ (**SAC**) of RESTful API because they do not constrain the resource response fields. Thereby leaving the implementers frustrating their consuming application engineers and/or frustrating users of the application due to performance issues.
+
+There are clear benefits of the **SAC**. According to Roy Fielding
+> "REST provides a set of architectural constraints that, when applied as a whole, emphasizes scalability of component interactions, the generality of interfaces, independent deployment of components, and intermediary components to reduce interaction latency, enforce security, and encapsulate legacy systems"
+
+However, there is no _constraint_  on how to limit or include additional fields for `resource` response. 
+For example, techniques "Sparse Fields", "Compound Documents" and "Pagination" are not defined in the **SAC**. 
+This leaves it up to the implementers to decide how, or even if, to implement these features.
+
+This is where [JSON:API](https://jsonapi.org/format/) and [GraphQL](https://graphql.org/learn/) (among a few others) 
+provide the needed additional constraints that address the previously listed "techniques" that RESTful API does not address. 
+There are many features that JSON API and GraphQl are capable of, but we are only focusing on the ones that achieve the goal of reducing the size and number of API requests that go between the client and server.
+
+## Adding Constraints through Specifications
+
+### JSON:API
+**JSON:API** has features such as **Sparse Fields**, **Compound Documents**, **Pagination**, and **Caching**.
+
+#### Sparse Fieldset
+This is the bread and butter that allows the request to specify the fields needed.
+There are more examples on Sparse Fieldsets at [jsonapi.org](https://jsonapi.org/examples/#sparse-fieldsets)
+```
+GET /books?fields[books]=author,themes 
+```
+
+#### Compound Documents
+This feature allows for including related resources with the initial request
+```
+GET /books?fields[books]=author,themes,&include=chapter1/title 
+```
+
+#### Caching
+JSON:API takes advantage of the caching ability that HTTP has built in. This reduces the need to return data that has not changed, saving bandwidth.
+
+#### Pagination
+JSON:API handles Pagination with a standard that `links` such as: `first`, `last`, `next`, and `previous` will return subnets of sort information.
+https://jsonapi.org/format/#fetching-pagination
+
+---
+
+### GraphQL
+**GraphQL** has features such as **Sparse Fields**, **Compound Documents**, **Pagination**, and **Caching**.
+
```

Comment:
Something like

As stated by Paul _____ (citation[https://blog.runscope.com/posts/you-might-not-need-graphql]) and ____ ((citation)[https://apisyouwonthate.com/blog/understanding-rpc-rest-and-graphql]), GraphQL is more similar to a Remote Procedure Call API  than to a standard REST-style API like JSON:API.